File: 1-Bin2dec/pyton/Bin2dec.py

def bin_to_dec(bin_str):
    dec_val = 0
    bin_length = len(bin_str)

    for i in range(bin_length):
#ikinin üssünü kullanarak ondalık eşdeğer hesaplıyor
        dec_val += int(bin_str[i]) * (2 ** (bin_length - i - 1))

    return dec_val

def main():
    try:
#binary girişini alıyoruz
        bin_input = input("Enter a binary number (max 8 digits) : ")

#girilen şeyin binary olup olmadığına bakıyoruz
        if not all(bit in '01' for bit in bin_input) or len(bin_input) > 8:
            raise ValueError("Invalid input. Please enter a binary number.")

#binary sayıyı decimal sayıya dönüştürüyoruz
        dec_result = bin_to_dec(bin_input)
        print(f"{bin_input} in decimal is = {dec_result}")
    
    except ValueError as e:
        print(f"error : {e}")

if __name__ == "__main__":
    main()    

File: 1-Bin2dec/java/Bin2dec/Bin2dec.java
import java.util.Scanner;

public class Bin2dec{

    public static int binToDec(String binStr) {       
    
    int decValue = 0;
    int binLength = binStr.length();

    for(int i = 0; i < binLength; i++) {

// ikinin üssünü kullanarak ondalık değerini buluyoruz

        decValue += Character.getNumericValue(binStr.charAt(i)) * Math.pow(2, (binLength - 1 - i));
        }  
    return decValue;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        try {
//kullanıcıdan sadece 0 veya 1 girmesini istiyoruz(binary 2 tabanında)
            System.out.println("Enter binary number(max 8): ");
            String binInput = scanner.next();

//gelen verinin binary olup olmadığına bakıyoruz, değilse hata fırlatıyoruz
            if (!binInput.matches("[01]+") || binInput.length() > 8) {
                throw new IllegalArgumentException("Invalid binary input.");
            }

//binary sayıyı decimal sayıya dönüştürüyoruz
            int decValue = binToDec(binInput);
            System.out.println(binInput + " in decimal is " + decValue);
                
            }catch (IllegalArgumentException e) {
                System.out.println("error: " +e.getMessage());
            }finally {
                scanner.close();
        }

    }
}
